orchestrator-salt/src/main/resources/salt/salt/postgresql/init.sls

```diff
@@ -24,14 +26,29 @@ init-services-db-remote:
 
 init-db-with-utf8:
   cmd.run:
-    - name: rm -rf /var/lib/pgsql/data && runuser -l postgres sh -c 'initdb --locale=en_US.UTF-8 /var/lib/pgsql/data > /var/lib/pgsql/initdb.log' && rm /var/log/pgsql_listen_address_configured
+    - name: rm -rf {{ postgres_data_directory }} && runuser -l postgres sh -c 'initdb --locale=en_US.UTF-8 {{ postgres_data_directory }} > /var/lib/pgsql/initdb.log' && rm /var/log/pgsql_listen_address_configured
     - unless: grep -q UTF-8 /var/lib/pgsql/initdb.log
 
+{%- if postgres_data_on_attached_disk %}
+
+change-db-location:
```

Comment:
Take a look at https://github.com/hortonworks/cloudbreak/blob/master/freeipa/src/main/resources/freeipa-salt/salt/freeipa/services.sls on how you can figure out service unit files.